### PR TITLE
publish

### PR DESCRIPTION
**Pull request by yossi-lavi**

### CODE REVIEW

File: .github/workflows/mvn-publish-all.yml [hunk 1]
```diff
@@ -1,31 +1,127 @@
-name: Publish All to maven
+name: Publish all to maven
 on:
-  workflow_dispatch:
+  workflow_call:
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:3]: We can remove the entire `workflow_call` event, as we never call this workflow from another workflow.

File: .github/workflows/mvn-publish-all.yml [hunk 2]
```diff
@@ -1,31 +1,127 @@
...
+
+      - name: Generate openapi
+        run: |
+          source VERSION
+          cd sdk
+          # update the generator config file with the required maven version
+          sed -i "s/artifactVersion: \"$MAVEN_VERSION/artifactVersion: \"${{ inputs.maven_version }}/g" config.yaml
+          cat config.yaml
+          # update the makefile with the required vault version
+          sed -i "s/VAULT_VERSION\s:=\s$MAVEN_VERSION/VAULT_VERSION\s:=\s${{ inputs.maven_version }}/g" Makefile
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:63]: I wouldn't change the Makefile with `sed`.
Instead, we should pass any variable to any make command we want.

File: .github/workflows/mvn-publish-all.yml [hunk 3]
```diff
@@ -1,31 +1,127 @@
...
+          cat Makefile
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:64]: Remove

File: .github/workflows/mvn-publish-all.yml [hunk 4]
```diff
@@ -1,31 +1,127 @@
...
+          # generate Java sdk for the openapi
+          make generate-openapi
+          sudo git apply patches/*.patch
+          cat openapi/pom.xml
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:68]: Remove

File: .github/workflows/mvn-publish-all.yml [hunk 5]
```diff
@@ -1,31 +1,127 @@
...
+          cd sdk
+          # update the generator config file with the required maven version
+          sed -i "s/artifactVersion: \"$MAVEN_VERSION/artifactVersion: \"${{ inputs.maven_version }}/g" config.yaml
+          cat config.yaml
+          # update the makefile with the required vault version
+          sed -i "s/VAULT_VERSION\s:=\s$MAVEN_VERSION/VAULT_VERSION\s:=\s${{ inputs.maven_version }}/g" Makefile
+          cat Makefile
+          # generate Java sdk for the openapi
+          make generate-openapi
+          sudo git apply patches/*.patch
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:67]: I'll call here `sudo chmod -R 664 sdk/openapi`.
Then, you don't need `sudo` for the git apply command or any other command.
Maybe just to better understand what is going on here do a `ls -l sdk` so we can see user, group & permissions for this folder.

File: .github/workflows/mvn-publish-all.yml [hunk 6]
```diff
@@ -1,31 +1,127 @@
...
+          cat openapi/pom.xml
+
+      - name: Build and publish openapi
+        run: |
+          sudo chmod -R 777 sdk/openapi
+          cd sdk/openapi
+          mvn clean deploy -B -P sign-artifacts -DskipTests
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:74]:
```suggestion
        working-directory: sdk/openapi
        run: mvn clean deploy -B -P sign-artifacts -DskipTests
```

File: .github/workflows/mvn-publish-all.yml [hunk 7]
```diff
@@ -1,31 +1,127 @@
...
+        env:
+          OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+          OSSRH_TOKEN: ${{ secrets.OSSRH_TOKEN }}
+          MAVEN_GPG_PASSPHRASE: ${{ secrets.MAVEN_GPG_PASSPHRASE }}
+
+      - name: Update version of client
+        run: |
+          # Updating version of client pom.xml
+          cd sdk/client
+          mvn versions:set -DnewVersion=${{ inputs.maven_version }}
```
Comment (ori-shalom) [inline .github/workflows/mvn-publish-all.yml:84]:
```suggestion
      - name: Update version of client pom.xml
        working-directory: sdk/client
        run: mvn versions:set -DnewVersion=${{ inputs.maven_version }}
```

File: .github/workflows/mvn-publish-all.yml (hunk 2 above)
Comment (yossi-lavi) [reply .github/workflows/mvn-publish-all.yml:63]: I'll deal with the make file in another PR. I'll fix also some problems @nirhaas raised

**Pull request merged**